openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -86,11 +88,99 @@ def state_swap_eigen_component(x: str, y: str, sign: int = 1, angle: float=0):
     return component
 
 
+@cirq.value_equality(approximate=True)
+class FermionicSimulationGate(cirq.EigenGate):
```

Comment:
I really don't want to force subclasses to implement EigenGate. Either implement the EigenGate methods here  in this class, or don't subclass it.

Reply:
Done. 

Reimplemented `EigenGate.__init__` in `FermionicSimulationGate.__init__` (just saving arguments as attributes) and subclassed `EigenGate` in each of the children.